index.js

```diff
@@ -0,0 +1,127 @@
+require('dotenv').config();
+const express = require("express");
+// const helmet = require("helmet");
+const bcrypt = require("bcryptjs");
+const dbFuncs = require("./dbFunctions");
+const jwt = require("jsonwebtoken");
+const morgan = require("morgan");
+// const KnexSessionStore = require("connect-session-store")(session);
+const server = express();
+
+// server.use(helmet());
+server.use(express.json());
+server.use(morgan())
+
+
+server.get("/", (req, res) => {
+  res.status(200).send("<h1>The server is up</h1>");
+});
+
+server.post("/users/register", async (req, res) => {
+  const userInfo = req.body;
+  const hash = bcrypt.hashSync(userInfo.password, 12);
+  userInfo.password = hash;
+
+  try {
+    const response = await dbFuncs.addUser(userInfo);
+    res.status(201).json(response);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json(err);
+  }
+});
+
+function generateToken(user) {
+	const payload = {
+		username: user.username,
+		name: user.name
+	};
+
+	const secret = process.env.JWT_SECRET;
+
+	const options = {
+		expiresIn: '2m',
+	};
+
+	return jwt.sign(payload, secret, options);
+}
+
+server.post("/login", async (req, res) => {
+  const creds = req.body;
+  try {
+    const user = await dbFuncs.getUser(creds);
+    if (user && bcrypt.compareSync(creds.password, user.password)) {
+        const token = generateToken(user);
+        res.status(200).json({ message: `Welcome ${user.name}`,token });
+    } else {
+      res
+        .status(401)
+        .json({ message: `Username and/or password are incorrect` });
+    }
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({ message: `Server error`, error: err });
+  }
+});
+
+server.get("/users", async (req, res) => {
+  try {
+    const users = await dbFuncs.getUsers();
+    res.status(200).json(users);
```

Comment:
You could save all of you status codes to a status code object to avoid the 'magic numbers' Luis talked about.